content/engine/release-notes/25.0.md

```diff
@@ -0,0 +1,102 @@
+- containerd image store: Add support for referencing images via the truncated ID with `sha256:` prefix. [moby/moby#46435](https://github.com/moby/moby/pull/46435)
+- containerd image store: Fix `docker images` showing intermediate layers by default. [moby/moby#46423](https://github.com/moby/moby/pull/46423)
+- containerd image store: Fix checking if the specified platform exists when getting an image. [moby/moby#46495](https://github.com/moby/moby/pull/46495)
+- containerd image store: Fix errors when multiple `ADD` or `ADD` instructions were used with the classic builder. [moby/moby#46383](https://github.com/moby/moby/pull/46383)
+- containerd image store: Fix stack overflow errors when importing an image. [moby/moby#46418](https://github.com/moby/moby/pull/46418)
+- containerd image store: Improve `docker pull` progress output. [moby/moby#46412](https://github.com/moby/moby/pull/46412)
+- containerd image store: Print the tag, digest, and size after pushing an image. [moby/moby#46384](https://github.com/moby/moby/pull/46384)
+- containerd image store: Remove panic from `UpdateConfig`. [moby/moby#46433](https://github.com/moby/moby/pull/46433)
+- containerd image store: Return an error when an image tag resembles a digest. [moby/moby#46492](https://github.com/moby/moby/pull/46492)
+
+### Packaging updates
+
+- Upgrade API to v1.44. [moby/moby#45468](https://github.com/moby/moby/pull/45468)
+- Upgrade Compose to `2.24.1`. [docker/docker-ce-packaging#980](https://github.com/docker/docker-ce-packaging/pull/980)
+- Upgrade containerd to v1.7.6. [moby/moby#46486](https://github.com/moby/moby/pull/46486)
+- Upgrade go to v1.20.8. [docker/cli#4557](https://github.com/docker/cli/pull/4557), [moby/moby#46453](https://github.com/moby/moby/pull/46453)
+- Upgrade runc to v1.1.9. [moby/moby#46192](https://github.com/moby/moby/pull/46192)
+- Upgrade swarmkit to v2.0.0. [docker/cli#4408](https://github.com/docker/cli/pull/4408), [moby/moby#46184](https://github.com/moby/moby/pull/46184)
+
+### Removed
+
+- API: Remove VirtualSize field for the `GET /images/json` and `GET /images/{id}/json` endpoints. [moby/moby#45469](https://github.com/moby/moby/pull/45469)
+- Remove deprecated `devicemapper` storage driver. [moby/moby#43637](https://github.com/moby/moby/pull/43637)
+- Remove deprecated orchestrator options. [docker/cli#4366](https://github.com/docker/cli/pull/4366)
+- Remove support for Debian Upstart init system. [moby/moby#45548](https://github.com/moby/moby/pull/45548), [moby/moby#45551](https://github.com/moby/moby/pull/45551)
+- Remove the `--oom-score-adjust` daemon option. [moby/moby#45484](https://github.com/moby/moby/pull/45484)
+- Remove warning for deprecated `~/.dockercfg` file. [docker/cli#4281](https://github.com/docker/cli/pull/4281)
+
+### Deprecated
+
+- Deprecate `IsAutomated` field and `is-automated` filter for `docker search`. [Deprecation notice](../deprecated.md#isautomated-field-and-is-automated-filter-on-docker-search)
```

Comment:
We also must include the API deprecation; https://github.com/docker/cli/pull/4796

There may be a few more; although (some) of those may be "Go" related, not directly user-facing; https://github.com/docker/cli/pulls?q=is%3Apr+milestone%3A25.0.0+label%3Aimpact%2Fdeprecation+is%3Aclosed

Or look at the `deprecated.md` which may contain more for v25 (deprecated, or removed in); https://github.com/docker/cli/blob/master/docs/deprecated.md

And for the API; https://github.com/moby/moby/blob/master/docs/api/version-history.md#v144-api-changes

Reply:
Added the logentries one too